packages/beta/src/api/templateGroups/templateGroupsApi.ts

```diff
@@ -0,0 +1,96 @@
+   *
+   * Create new Template Groups. The user creating a Template Group is automatically added as an owner of that Template Group,
+   * however more owners can optionally be added by setting the 'owners' field.
+   *
+   * ```js
+   * const templateGroups = [
+   *   { externalId: 'Wells', description: 'Models a Well system', owners: ['user.name@example.com'] },
+   * ];
+   * const createdTemplateGroups = await client.templateGroups.create(templateGroups);
+   * ```
+   */
+  public create = (
+    items: ExternalTemplateGroup[]
+  ): Promise<TemplateGroup[]> => {
+    return super.createEndpoint(items);
+  };
+
+  /**
+   * [Upsert Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroupsUpsert)
+   *
+   * Create or update new Template Groups. The user creating a Template Group is automatically added as an owner of that Template Group,
+   * however more owners can optionally be added by setting the 'owners' field.
+   *
+   * ```js
+   * const templateGroups = [
+   *   { externalId: 'Wells', description: 'Models a Well system', owners: ['user.name@example.com'] },
+   * ];
+   * const upsertedTemplateGroups = await client.templateGroups.upsert(templateGroups);
+   * ```
+   */
+  public upsert = (
+    items: ExternalTemplateGroup[]
+  ): Promise<TemplateGroup[]> => {
+    return super.createEndpoint(items, this.url('upsert'));
+  };
+
+  /**
+   * [Retrieve Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroupsByids)
+   * <!-- or [similar](https://doc.cognitedata.com/api/v1/#operation/getEventByInternalId) -->
+   *
+   * Retrieve Template Groups by externalIds.
+   *
+   * ```js
+   * const templateGroups = await client.templateGroups.retrieve([{externalId: 'abc'}]);
+   * ```
+   */
+  public retrieve = (ids: ExternalId[], ignoreUnknownIds: Boolean = false) => {
+    return super.retrieveEndpoint(ids, { ignoreUnknownIds });
+  };
+
+  /**
+   * [List Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroupsList)
+   *
+   * List Template Groups or filter them by owner.
+   *
+   * ```js
+   * const templateGroups = await client.templateGroups.list({ filter: { owners: ["user.name@example.com"] } });
+   * ```
+   */
+  public list = (
+    query?: TemplateGroupFilterQuery
+  ): CursorAndAsyncIterator<TemplateGroup> => {
+    return super.listEndpoint(this.callListEndpointWithPost, query);
+  };
+
+  /**
+   * [Delete Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroupsDelete)
+   *
+   * Delete Template Groups by externalIds.
+   *
+   * ```js
+   * await client.templateGroups.delete([{ externalId: "Wells" }]);
+   * ```
+   */
+  public delete = (ids: ExternalId[]) => {
+    return super.deleteEndpoint(ids, {
+      ignoreUnknownIds: true,
```

Comment:
for some reason it's by default only for assets? need to recheck this, maybe it worth being consistently bad :)

Reply:
Hmm, so I should rather expose that to the user then?
For events we are not exposing it at all, so it'll be inconsistent anyways.